### DISCUSSION TITLE
Anyone had success attaching a debugger to their external scanner?

### DISCUSSION THREAD

**Discussion by ahelwer** [tree-sitter/tree-sitter#1620] [Q&A] [answered]
Pretty much what's in the title. I use VS Code on Windows but as long as I can set a breakpoint somehow and step through my external scanner code I'll be happy.

**Comment by sogaiu** [comment]
I think at one point I compiled with debug info and examined things under gdb but it's been a while.

**Comment by maxbrunsfeld** [comment, marked as answer]
I've often used debuggers when developing external scanners. If you're using the `tree-sitter` CLI, you can definitely spawn that under the debugger; you just need to grab the native executable, not the JavaScript wrapper script. With LLDB, it's a similar workflow to any other C/C++ program. I'm not very familiar with the windows debugging tools.
